op-service/espresso/nmt.go

```diff
@@ -0,0 +1,22 @@
+package espresso
+
+import (
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+// This function mocks batch transaction validatijon against a set of valid HotShot headers.
+// It pretends to verify that the set of transactions (txns) in a batch correspond to a set of n NMT proofs
+// (p1, ... pn) against headers h1,...hn.
+//
+// We assume that his function makes an external call to fetch the block headers beteen firstBlock and lastBlock.
```

Comment:
I think it would be cleaner to have the caller pass in all the headers, which they would get by calling a second method on the `HotShotContractProvider`, like `GetHeader(i)` or something.